processor/elasticinframetricsprocessor/factory.go

```diff
@@ -0,0 +1,47 @@
+package elasticinframetricsprocessor // import "github.com/elastic/opentelemetry-collector-components/processor/elasticinframetricsprocessor"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/processor"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+)
+
+const (
+	typeStr = "elasticinframetrics"
+)
+
+var processorCapabilities = consumer.Capabilities{MutatesData: true}
+
+// NewFactory returns a new factory for the Filter processor.
+func NewFactory() processor.Factory {
+	return processor.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		processor.WithMetrics(createMetricsProcessor, component.StabilityLevelAlpha),
```

Comment:
If we think the configuration/logic is still in a very early stage, we could use the `Development` stability level instead.

Reply:
If I see the difference between [Development](https://github.com/open-telemetry/opentelemetry-collector/tree/main?tab=readme-ov-file#development) and [Alpha](https://github.com/open-telemetry/opentelemetry-collector/tree/main?tab=readme-ov-file#alpha), I feel alpha to be more suitable. 
Although a comment of "it might not be available as part of any distributions yet" in development is true for us. 
But since this is in Elastic repo, I think we should be good with Alpha. 
Since we will be asking some users to use this processor soon.

Reply:
Sounds good! We can still do breaking changes `The configuration options might change often without backwards compatibility guarantees.` Thanks for the clarification